manifests/init.pp

```diff
@@ -287,6 +301,8 @@
   Boolean $manage_logwatch               = true,
   String $operator_command               = "/usr/sbin/bsmtp -h localhost -f bacula@${::fqdn} -s \\\"Bacula Intervention Required (for %c)\\\" %r",
   String $plugin_dir                     = '/usr/lib64/bacula',
+  Optional[Array] $storage_hash           = undef,
```

Comment:
This should be set to Optional[Hash] $storage_hash to avoid puppet failures.  Running this code on our director results in an error as follows.

Error: Could not retrieve catalog from remote server: Error 500 on SERVER: Server Error: Evaluation Error: Error while evaluating a Resource Statement, Class[Bacula::Storage]: parameter 'storage_hash' expects an Array value, got Hash (file: /etc/puppetlabs/code/environments/bacula_test/forge_modules/bacula/manifests/init.pp, line: 391, column: 5) on node bacula.example.com

Reply:
Good catch. I forgot to adjust the default value to an empty array.
I used a hash before until realising that that wont work.

Reply:
No problem.  Once thing I've noticed is that the default value for Block Checksum now evaluates as "yes" even when I set the parameter to false in hiera.  I believe this might because there is no "false" value set in the selector.

```
   'Block Checksum'   => $block_checksum ? {
      default => yes,
      no      => no,
    },
```

This should most likely be set as follows.

```
   'Block Checksum'   => $block_checksum ? {
      default => yes,
      false      => no,
    },
```

Puppet does not treat yes/no as "truthy" values, they are simply strings.

Reply:
This module previously treated yes/no values for bacula settings as strings however I converted it to use booleans for each parameter instead.  The puppet templates then evaluate these booleans to fill in the proper yes/no values.

Reply:
I've fixed this with commit 9c0a1e2